lib/elixir/src/elixir_erl_compiler.erl

```diff
@@ -75,11 +75,12 @@ handle_file_warning(_, File, {Line, Module, Desc}) ->
 %% Handle warnings
 
 handle_file_error(File, {beam_validator, Rest}) ->
-  Message = beam_validator:format_error(Rest),
-  elixir_errors:compile_error([{line, 0}], File, elixir_utils:characters_to_binary(Message));
+  elixir_errors:form_error([{line, 0}], File, beam_validator, Rest);
 handle_file_error(File, {Line, Module, Desc}) ->
-  Message = format_error(Module, Desc),
-  elixir_errors:compile_error([{line, Line}], File, elixir_utils:characters_to_binary(Message)).
+  elixir_errors:form_error([{line, Line}], File, ?MODULE, {Module, Desc}).
```

Comment:
Oh but that sucks 😄 If I'm correct this is the only place where we still use `compile_error`. I would push for finding a better way to abstract this over `format_error/1`. What about 

```elixir
form_error(Meta, File, ?MODULE, {erlang_error, Module, Desc})
```

?